### PR TITLE
frame combination step function and test

### PR DESCRIPTION
**Pull request by semaphoreP**
## Describe your changes

New step function to combine subsets of frames together. Made currently for the intent of the flat field e2e to combine subexposures of a raster scan. I started a new combine.py since I anticipate we want to combine frames in different frames (e.g., collapse via keywords, sigma clipping, etc). 

I also updated the recording of parent filenames to include grand-parents. This way, a file (E.g., flat field) would record its parents, but also the frames that were combined together from its parents. In the current logic, great grand parents of arbitrary depth are also tracked. 

## Type of change

Please delete options that are not relevant (and this line).

- New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have linted my code
- [x] I have verified that all unit tests pass in a clean environment and added new unit tests, as needed
- [x] I have verified that all docstrings are properly formatted and added new documentation, as needed

### CODE REVIEW

File: corgidrp/combine.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+"""
+Module to support frame combination
+"""
+import numpy as np
+import corgidrp.data as data
+
+
+def combine_images(data_subset, err_subset, dq_subset, collapse="lower"):
+    """
+    Combines several images together
```
Comment (maxwellmb) [inline corgidrp/combine.py:10]: It looks to me like the operations below try to maintain the total number of counts, which isn't the intuitive mean or median behavior that you'd expect. I think we might want the option to multiply by tot_frames or not and to describe that more in the docstring.

File: corgidrp/combine.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        err_subset = np.copy(input_dataset.all_err[num_frames_per_group*i:num_frames_per_group*(i+1)])
+        dq_subset = input_dataset.all_dq[num_frames_per_group*i:num_frames_per_group*(i+1)]
+
+        data_collapse, err_collapse, dq_collapse = combine_images(data_subset, err_subset, dq_subset, collapse=collapse)
+
+        # grab the headers from the first frame in this sub sequence
+        pri_hdr = input_dataset[num_frames_per_group*i].pri_hdr.copy()
+        ext_hdr = input_dataset[num_frames_per_group*i].ext_hdr.copy()
+        err_hdr = input_dataset[num_frames_per_group*i].err_hdr.copy()
+        dq_hdr = input_dataset[num_frames_per_group*i].err_hdr.copy()
```
Comment (maxwellmb) [inline corgidrp/combine.py:82]: Should we also be grabbing the hdu_list if it exists?

**Review by maxwellmb — CHANGES_REQUESTED** [review]
Overall this looks fine, the biggest issue is that the combination is much closer to a sum than it is a mean or median and this needs to be clearer, either with different keyword values to `collapse`, and/or a clearer docstring, in both the combine images function and the step function.

File: corgidrp/combine.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+"""
+Module to support frame combination
+"""
+import numpy as np
+import corgidrp.data as data
+
+
+def combine_images(data_subset, err_subset, dq_subset, collapse="lower"):
```
[the remark this answers is not in the document]
Comment (semaphoreP) [reply corgidrp/combine.py]: oops typo. fixed

File: corgidrp/combine.py (hunk 1 above)
Comment (semaphoreP) [reply corgidrp/combine.py:10]: I added a second keyword to toggle the scaling by tot_frames

File: corgidrp/combine.py (hunk 2 above)
Comment (semaphoreP) [reply corgidrp/combine.py:94]: We were doing it right below, but I've moved it up here to be more explicit.

**Comment by semaphoreP** [comment]
@maxwellmb ready for re-review

File: corgidrp/combine.py (hunk 2 above)
Comment (maxwellmb) [reply corgidrp/combine.py:94]: Whoops, I missed it, my bad!

**Review by maxwellmb — APPROVED** [review]

**Pull request merged**